File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.java
class Solution {
    public List<String> stringMatching(String[] words) {
        Set<String> st = new HashSet<>();
        List<String> ans = new ArrayList<>();
        int n = words.length;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i != j) {
                    if (words[i].contains(words[j]))
                        st.add(words[j]);
                }
            }
        }

        for (String str : st)
            ans.add(str);
        return ans;
    }
}

File: 1524-string-matching-in-an-array/1524-string-matching-in-an-array.cs
public class Solution {
    public IList<string> StringMatching(string[] words) {
        IList<string> ans = new List<string>();
        var st = new HashSet<string>();
        int n = words.Length;

        for (var i = 0; i < n; ++i) {
            for (var j = 0; j < n; ++j) {
                if (i != j) {
                    if (words[i].Contains(words[j]))
                        st.Add(words[j]);
                }
            }
        }
        foreach (var str in st)
            ans.Add(str);
        return ans;
    }
}